bundles/org.openhab.binding.tapocontrol/src/main/java/org/openhab/binding/tapocontrol/internal/TapoDiscoveryService.java

```diff
@@ -58,41 +57,29 @@ public class TapoDiscoveryService extends AbstractDiscoveryService implements Th
      * INIT CLASS
      */
     public TapoDiscoveryService() {
-        super(SUPPORTED_THING_TYPES_UIDS, TAPO_DISCOVERY_TIMEOUT_S, false);
+        super(TapoBridgeHandler.class, SUPPORTED_THING_TYPES_UIDS, TAPO_DISCOVERY_TIMEOUT_S, false);
     }
 
     /**
      * activate
      */
     @Override
-    public void activate() {
-        TapoBridgeConfiguration config = bridge.getBridgeConfig();
+    public void initialize() {
+        thingHandler.setDiscoveryService(this);
+        TapoBridgeConfiguration config = thingHandler.getBridgeConfig();
```

Comment:
I think it should also call `super.initialize();` here?

Reply:
Debatable, because the service handles its background discovery by itself and the `activate` method didn't call `super.activate` either. I have refactored that to use the standard mechanism, unfortunately I can't test it.